src/proc.cpp

```diff
@@ -574,6 +580,11 @@ void remove_disowned_jobs(job_list_t &jobs) {
     while (iter != jobs.end()) {
         const auto &j = *iter;
         if (j->flags().disown_requested && j->job_chain_is_fully_constructed()) {
+            for (auto &process : j->processes) {
```

Comment:
See my comment in the conversation, but briefly, can we wrap this in a `if (j->pgid() == shell_pgid)` precondition?

Reply:
I've decided to refactor this all into one spot.